hybrid/core.py

```diff
@@ -313,6 +313,12 @@ def error(self, exc):
         """
         raise exc
 
+    def halt(self):
+        """Called by `stop()`. Override this method (instead of `stop`) to handle
+        stopping of one blocking call of `next`. Defaults to NOP.
+        """
```

Comment:
"""Terminate a blocking call of `next`.

Called by `stop`. Override this method, instead of `stop`, to terminate an iteration of an instantiated :class:`Runnable`.
Defaults to NOP.
"""